science/apbs/Portfile

```
@@ -20,64 +17,86 @@ long_description        APBS is a software package for the numerical \
+# Follow Debian's approach of not building those
+# features like BEM, GEOFLOW or PBAM that only
+# exist in apbs's github without tagged releases yet.
+configure.args-append   -DAPBS_STATIC_BUILD=ON \
+                        -DBUILD_SHARED_LIBS=OFF \
+                        -DBUILD_TOOLS=ON \
+                        -DENABLE_OPENMP:BOOL=OFF \
+                        -DENABLE_BEM=OFF \
+                        -DENABLE_GEOFLOW=OFF \
+                        -DENABLE_PBAM=OFF \
+                        -DBUILD_MALOC=OFF \
+                        -DBUILD_SUPERLU=OFF \
+                        -DBUILD_SHARED_LIBS=ON \
+                        -DFETCHCONTENT_SOURCE_DIR_FETK=${worksrcpath}/FETK-1.9.3 \
+                        -DCMAKE_C_COMPILER_ARG1:STRING="${configure.cflags} -I${worksrcpath}/include" \
+                        -DCMAKE_CXX_COMPILER_ARG1:STRING="${configure.cxxflags} -I${worksrcpath}/include -stdlib=libc++" \
+                        -DSUPERLU_INCLUDES=${prefix}/include -DSUPERLU_LIBRARIES=${prefix}/lib 
+
+patchfiles              cmake-maloc.patch disable_python_cmake_check.patch patch-testfiles.patch open-U-obsolete.patch python3.patch wrong-path-for-interpreter.patch
 test.run  yes
 test {
     ln ${worksrcpath}/tools/manip/inputgen.py ${worksrcpath}/tests/
     ln ${worksrcpath}/tools/manip/psize.py    ${worksrcpath}/tests/
 
-    if {[mpi_variant_isset]} {
-        reinplace "s|mpiexec|\"${mpi.exec}\", \"-n\", \"${build.jobs}\"|" ${worksrcpath}/tests/apbs_tester.py
+    platform darwin {
+       if {${os.major} < 13} {
+          system -W ${worksrcpath}/tests "PATH=${destroot}${prefix}/bin:/usr/bin:/bin:${prefix}/bin ${prefix}/bin/python3.12 apbs_tester.py -c test_cases.cfg || exit 0"
+       } else {
+          system -W ${worksrcpath}/tests "PATH=${destroot}${prefix}/bin:/usr/bin:/bin:${prefix}/bin python3 apbs_tester.py -c test_cases.cfg || exit 0"
+       }
     }
-    
-    system -W ${worksrcpath}/tests "python apbs_tester.py"
     system -W ${worksrcpath}/tests "cat test.log"
     system -W ${worksrcpath}/tests "if grep FAILED test.log; then echo \"FAIL\"; exit 1; fi"
 }
 
-# Some cmake tests will fail with "error: unrecognized option '-arch'" from gcc 4.6 and earlier. Same with dragonegg based on it.
-# llvm-gcc-4.2 will fail to build with: error: unrecognized command line option "-std=c++0x"
-# It comes with clang for OSX 10.6 so we need to blacklist clang to prevent it being used.
-compiler.blacklist      macports-gcc-4.4 macports-gcc-4.5 macports-gcc-4.6 \
-    macports-dragonegg-3.4 gcc-4.2 llvm-gcc-4.2 apple-gcc-4.2 macports-llvm-gcc-4.2 {clang < 425} \
-    macports-clang-3.4
-# g++ will just fail to link apbs_geoflow. probably about -lstdc++ stuff. So, just use Fortran.
-compilers.choose        fc
-mpi.setup
+pre-patch {
+    system "mv ${worksrcpath}/../FETK-1.9.3 ${worksrcpath}/FETK-1.9.3"
+}
```

Please use tcl commands where available instead of calling out to a separate executable with `system`.

You probably want to do this at post-extract, not pre-patch, in case the patch phase needs to be repeated due to a failure.

`${worksrcpath}/..` is more simply known as `${workpath}`.

```suggestion
post-extract {
    move ${workpath}/FETK-1.9.3 ${worksrcpath}/FETK-1.9.3
}
```

---

Switched FETK-1.9.3 directory move to post-extract using the tcl move command.